pep_sphinx_extensions/pep_processor/transforms/pep_headers.py

```diff
@@ -97,20 +98,28 @@ def apply(self) -> None:
             field.parent.remove(field)
 
 
-def _list_name_from_thread(node: nodes.reference) -> nodes.raw:
+def _pretty_thread(text: nodes.Text) -> nodes.Text:
+    # we want to keep these as lowercase:
+    if "python-dev" in text:
+        return nodes.Text("python-dev")
+    if "python-committers" in text:
+        return nodes.Text("python-committers")
```

Comment:
```suggestion
```

> we want to keep these as lowercase

I am a little confused—is there a reason for this? And why these?

The canonical capitalization for Python-Dev on the [archives](https://mail.python.org/archives/list/python-dev@python.org/) and [main page](https://mail.python.org/mailman3/lists/python-dev.python.org/) is in title case, and so is every current instance this PR is replacing. Also, it would be inconsistent with the capitalization of the other lists, and is (IMO) slightly harder to scan at a glance and looks uglier. Plus, its a special case we don't have to worry about and extra code we don't have to maintain.

Reply:
> I am a little confused—is there a reason for this? And why these?

Because I most often see people refer to the list name specially as "python-dev", versus e.g. "Disutils SIG". We'd still need special casing to return "Python-Dev", as otherwise the hyphen would be replaced with a space.

> Python-Committers

I used this pattern `^[DR][a-zA-Z\-]+: .*/python-[^d]` to check for non-dev lists, committers is used for 634, 635, and 636. I'd prefer to reflect reality on the implementation and if we decide tightening rules is a good idea we can do so later.

A

Reply:
> Because I most often see people refer to the list name specially as "python-dev", versus e.g. "Disutils SIG".

Personally, I wouldn't think what an individual has anecdotally seen elsewhere is as relevant as what is actually currently used here by previous PEP authors for this header, which in every case was `Python-Dev` when referring to the list name, (never mind the other arguments for consistency, canonicility, readability and aesthetic preference), so I think we should just follow existing convention rather than adding bespoke code to special-case this specific list name. And why do it for Python-Dev, but not Python-Ideas? Or Python-List? Or ...? Special casing arbitrary lists opens up a rabbit hole that is best simply avoided.

> We'd still need special casing to return "Python-Dev", as otherwise the hyphen would be replaced with a space.

Not if we simply leave the hyphen as-is, as I suggest below.

> I'd prefer to reflect reality on the implementation and if we decide tightening rules is a good idea we can do so later.

The current reality is that as a rule in the existing Discussions-To header, as well as in most cases of their canonical names, the list names are capitalized (and aside from the closed Distutils-SIG, all contain hyphens in both places). Therefore, I'm not sure why we should add special-case code to arbitrarily change that for some lists and not others.

Reply:
I would also like to minimize the number of special cases. Those are difficult to maintain and difficult to predict for a PEP author or editor. It's fine to just say "Python-Dev".

Reply:
Will just leave title case normalisation then. 'Disutils Sig' seems slightly ugly but perhaps predictable.

A

Reply:
I can't speak for @JelleZijlstra , but I wasn't suggesting that particular change. "SIG" is makes sense as the one specific replacement to do since it is very common in mailing list names used for PEPs, and SIGs are specifically cited in PEP 1 as a good place to hold the PEP discussion. Scanning through, I'm not aware of any other common examples used in more than one PEP.

Reply:
Possibly because they were just using naive title-case capitalization rules and/or copying from the previous PEP(s) that did (all PEPs in that list that actually had a list name used the exact same string, verbatim). Its not super important, but making it all caps also makes it clear that it is the acronym SIG, and not, e.g. an abbreviation for one of many common words that starts with "Sig" (significant, signal, sign, etc).

Reply:
That makes sense too, I think let's just make @AA-Turner make the decision here.